lib/internal/test_runner/harness.js

```diff
@@ -33,8 +33,14 @@ const testResources = new SafeMap();
 
 testResources.set(reporterScope.asyncId(), reporterScope);
 
-function createTestTree(options = kEmptyObject) {
-  return setup(new Test({ __proto__: null, ...options, name: '<root>' }));
+function createTestTree({ coverage, coverageExcludeGlobs, coverageIncludeGlobs, ...options } = kEmptyObject) {
+  const coverageOpts = coverage == null ? undefined : {
+    __proto__: null,
+    coverage,
+    coverageExcludeGlobs,
+    coverageIncludeGlobs,
+  };
+  return setup(new Test({ __proto__: null, ...options, name: '<root>' }), coverageOpts);
```

Comment:
I think we should pass all provided options (not just coverage options) to `setup()` and store them on the harness object (I was planning to do that after some of the open PRs land).

The reason is that I want to get rid of all uses of `parseCommandLine()` except two. There should be one in `lib/internal/main/test_runner.js` for the CLI, and one in `lib/internal/test_runner/harness.js` for the case where a test file is run directly. That way, `run()` needs access to less global state. The use of `parseCommandLine()` in `lib/internal/test_runner/test.js` would be replaced by accessing `root.harness.config.FOO`.

Reply:
@cjihrig I implemented this, mostly works, I still have some issue with the `only` flag, I hipe to get to it by the end of the week.
Are my changes anywhere near what you had in mind? 🙂